### PR TITLE
Db 126 delete users

### PR DESCRIPTION
**Pull request by jworcestBAH**
This adds the backend route to delete a user for DB-126, frontend half of this will likely occur next sprint.

### CODE REVIEW

File: dataactbroker/README.md [hunk 1]
```diff
@@ -428,6 +428,26 @@ Example output:
...
 
+#### POST "/v1/delete_user/"
+Delete specified user. Calls to this route should include the key "email" to specify user to be deleted, route is only
+accessible for users with website_admin or agency_admin permissions.
+
+Example input:
+
+```json
+{
+    "email": user@agency.gov
```
Comment (jworcestBAH) [inline dataactbroker/README.md]: Good catch, fixed

File: dataactbroker/handlers/userHandler.py [hunk 2]
```diff
@@ -69,6 +70,19 @@ def getUserByUID(self,uid):
...
+    def deleteUser(self, email):
+        """ Delete user with specified email.  Submissions from that user are not deleted, instead they have
+        user id set to null  """
+
+        # Find user
+        user_query = self.session.query(User).filter(User.email == email)
+        user_to_be_deleted = self.runUniqueQuery(user_query, "No user with that email", "Multiple users with that email")
+        # Set all their submissions to have no user
+        self.session.query(Submission).filter(Submission.user_id == user_to_be_deleted.user_id).update(
+            values = {"user_id":None})
```
Comment (jworcestBAH) [inline dataactbroker/handlers/userHandler.py]: Good call, made this an explicit foreign key with set null and removed the now duplicated code.

File: tests/integration/userTests.py [hunk 3]
```diff
@@ -400,4 +400,40 @@ def test_email_users(self):
...
+        self.login_admin_user()
+        # Create user to be deleted, done within test to avoid interfering with tests on number of users
+        email = "to_be_deleted@agency.gov"
+        user_to_be_deleted = createUserWithPassword(email, "unused", Bcrypt())
+        # Give this user a submission
+        with createApp().app_context():
+            sess = GlobalDB.db().session
+            sub = Submission(user_id = user_to_be_deleted.user_id)
+            sub.reporting_start_date = datetime(2015, 10, 1)
+            sub.reporting_end_date = datetime(2015, 12, 31)
```
Comment (jworcestBAH) [inline tests/integration/userTests.py]: Changed to use the factory

File: tests/unit/dataactbroker/test_delete_user.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    database.session.add(user_to_be_deleted)
+    database.session.add(other_user)
+    database.session.commit()
+    # Create two submissions for this user and one for a different user
+    sub_one = SubmissionFactory(user_id = user_to_be_deleted.user_id)
+    sub_two = SubmissionFactory(user_id = user_to_be_deleted.user_id)
+    other_sub = SubmissionFactory(user_id = other_user.user_id)
+    database.session.add(sub_one)
+    database.session.add(sub_two)
+    database.session.add(other_sub)
```
Comment (jworcestBAH) [inline tests/unit/dataactbroker/test_delete_user.py]: Thanks, didn't know that was a thing

File: tests/unit/dataactbroker/test_delete_user.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    database.session.commit()
+    # Delete a user
+    user_handler.deleteUser(email)
+    # Confirm user has been deleted and that user's submissions have no user_id
+    assert(len(database.session.query(User).filter(User.email == email).all()) == 0)
```
Comment (jworcestBAH) [inline tests/unit/dataactbroker/test_delete_user.py]: Changed to use count and filter_by

File: tests/unit/dataactbroker/test_delete_user.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    assert(sub_one.user_id is None)
+    assert(sub_two.user_id is None)
+    # Confirm that other submission was not affected
+    assert(other_sub.user_id == other_user.user_id)
```
Comment (jworcestBAH) [inline tests/unit/dataactbroker/test_delete_user.py]: Good point, changed here and will use this syntax going forward.

**Comment by cmc333333** [comment]
Looks great!

**Pull request merged**